File: tests/typesafety/typescript/pure_clock.ts
// expect: passes

export function remaining(deadline: Date, now: Date): number {
  return deadline.getTime() - now.getTime();
}

const dispatch = new Date("2026-08-13T16:00:00");

export const before = remaining(dispatch, new Date("2026-08-13T15:30:00"));
export const after = remaining(dispatch, new Date("2026-08-13T16:45:00"));

File: tests/typesafety/python/pure_clock.py
# expect: passes
# run: yes
# title: remaining takes now as a parameter, so a test can pin its answer
# improves: impure_clock
from datetime import datetime, timedelta


def remaining(deadline: datetime, now: datetime) -> timedelta:
    return deadline - now


DISPATCH = datetime(2026, 8, 13, 16, 0)

assert remaining(DISPATCH, now=datetime(2026, 8, 13, 15, 30)) == timedelta(minutes=30)
assert remaining(DISPATCH, now=datetime(2026, 8, 13, 16, 45)) == timedelta(minutes=-45)
